specification/cognitiveservices/data-plane/SpellCheck/V1.0/SpellCheck.json

```diff
@@ -0,0 +1,468 @@
+              "name": "Mode",
+              "modelAsString": true
+            }
+          },
+          {
+            "name": "PreContextText",
+            "in": "formData",
+            "description": "A string that gives context to the text string. For example, the text string petal is valid. However, if you set preContextText to bike, the context changes and the text string becomes not valid. In this case, the API suggests that you change petal to pedal (as in bike pedal). This text is not checked for grammar or spelling errors. The combined length of the text string, preContextText string, and postContextText string may not exceed 10,000 characters. You may specify this parameter in the query string of a GET request or in the body of a POST request.",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "PostContextText",
+            "in": "formData",
+            "description": "A string that gives context to the text string. For example, the text string read is valid. However, if you set postContextText to carpet, the context changes and the text string becomes not valid. In this case, the API suggests that you change read to red (as in red carpet). This text is not checked for grammar or spelling errors. The combined length of the text string, preContextText string, and postContextText string may not exceed 10,000 characters. You may specify this parameter in the query string of a GET request or in the body of a POST request.",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "Text",
+            "in": "formData",
+            "description": "The text string to check for spelling and grammar errors. The combined length of the text string, preContextText string, and postContextText string may not exceed 10,000 characters. You may specify this parameter in the query string of a GET request or in the body of a POST request. Because of the query string length limit, you'll typically use a POST request unless you're checking only short strings.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success.",
+            "schema": {
+              "$ref": "#/definitions/SpellCheck"
+            }
+          },
+          "default": {
+            "description": "An error has occurred. Check the response type and/or status code for more details.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Successful query": {
+            "$ref": "./examples/SuccessfulSpellModeRequest.json"
+          },
+          "Successful Proof Mode query": {
+            "$ref": "./examples/SuccessfulProofModeRequest.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SpellCheck": {
```

Comment:
Please change this one to SpellCheck**Model** (to avoid a conflict with the operation ID)

(checked with @amarzavery)

Reply:
So "SpellCheck" appears to be the name of a schema in swagger that I only share. Would it be acceptable to change the operation ID instead?

Reply:
> So "SpellCheck" appears to be the name of a schema in swagger that I only share. 

Can you explain a bit more on this? 

The model name is just a logical name given to the schema so that customers can refer to it when using the SDK. It has no bearing on the REST API. If someone is using the REST API directly then they would not provide the model name but provide a sample based on the schema.

I can foresee that this could be a problem with discriminators. However, we have `"x-ms-discriminator-value": "SpellCheck"` to your rescue for that model definition.

Reply:
"The model name is just a logical name given to the schema so that customers can refer to it when using the SDK." -> I intended to say, I do not own this schema. :) I only use it. So I am hesitant to change this.

Reply:
as we discussed offline, I changed the name of the operationID